pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/controllers/controller.go

```diff
@@ -160,6 +174,17 @@ func (r *{{ .Resource.Kind }}Reconciler) Reconcile(ctx context.Context, req ctrl
 	if is{{ .Resource.Kind }}MarkedToBeDeleted {
 		if controllerutil.ContainsFinalizer({{ lower .Resource.Kind }}, {{ lower .Resource.Kind }}Finalizer) {
 			log.Info("Performing Finalizer Operations for {{ .Resource.Kind }} before delete CR")
+
+			// The following implementation will update the status
+			meta.SetStatusCondition(&{{ lower .Resource.Kind }}.Status.Conditions, metav1.Condition{Type: typeDegraded{{ .Resource.Kind }},
+				Status: metav1.ConditionTrue, Reason: "Finalizing",
```

Comment:
```suggestion
				, Reason: "Finalizing", Message: "Performing finalizer operations",
```

@NikhilSharmaWe same reasons here, since we are beginning to perform the finalizer operations we need to add it without the status. Therefore, before we set we need to ensure that we have NOT the status Downgrade added to the CR already. It means, we did not tried to do the finalizer operations and that fails.

Reply:
@NikhilSharmaWe the setStatus implementation already check if that exist or not so we just need to use the set directly without the need to check, see: https://github.com/kubernetes/apimachinery/blob/v0.24.3/pkg/api/meta/conditions.go#L30